core/matrix/csr.cpp

```diff
@@ -619,6 +623,16 @@ void Csr<ValueType, IndexType>::inv_scale_impl(const LinOp* alpha)
 }
 
 
+template <typename ValueType, typename IndexType>
+void Csr<ValueType, IndexType>::add_scaled_identity_impl(const LinOp* const a,
+                                                         const LinOp* const b)
+{
+    this->get_executor()->run(csr::make_add_diagonal_elems(this, false));
```

Comment:
Not 100% happy about this one, as it may cause issues e.g. if we are using views to application data.

Reply:
Yes, that's a good point. Maybe instead we should just check if all diagonals exist and throw an exception if a diagonal entry is missing?

Reply:
That's what I had in mind, yes